### PR TITLE
bot requirements corrections

### PR DESCRIPTION
**Pull request by derekgroh**
## Pull Request Checklist

Addresses bot Pull Requests #67 #95 #107

#### General

- [X] RuboCop passes

#### Purpose

Resolves outstanding bot requirements for updates to rubocop, rake and codeclimate.

### CODE REVIEW

File: bin/check-windows-process.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@
   opts.on('-p', '--processname PROCESS', 'Unique process string to search for.') do |p|
     options[:procname] = p
     if p == ''
-      STDERR.puts 'Empty string for -p : Expected a string to match against.'
+      warn 'Empty string for -p : Expected a string to match against.'
```
Comment (majormoses) [inline bin/check-windows-process.rb]: as you can see below it exits a 3 so this should be an unknown instead to maintain backwards compatibility.

File: bin/check-windows-process.rb [hunk 2]
```diff
@@ -53,7 +53,7 @@
     begin
       options[:warn] = Integer(w)
     rescue ArgumentError
-      STDERR.puts 'Optional -w needs to be a value in seconds'
+      warn 'Optional -w needs to be a value in seconds'
```
Comment (majormoses) [inline bin/check-windows-process.rb]: see previous comment.

File: bin/check-windows-process.rb [hunk 3]
```diff
@@ -62,7 +62,7 @@
 parser.parse!
 
 if options[:procname] == ''
-  STDERR.puts 'Expected a process to match against.'
+  warn 'Expected a process to match against.'
```
Comment (majormoses) [inline bin/check-windows-process.rb]: if we warn here the exception will be swallowed.

**Comment by derekgroh** [comment]
@majormoses Leave it for you to merge to see if there are any other items to address.

**Comment by majormoses** [comment]
> @majormoses Leave it for you to merge to see if there are any other items to address.

Sorry I am not sure I am following, I see my conversations marked as resolved but no code change. Are you wanting me to update the code based on the comments I put?

**Comment by nixwiz** [comment]
@majormoses can you review and merge now?  Once this is done, I'd like to get some of the latest dependencies brought in if they aren't addressed here and then work to add Bonsai support.

**Comment by majormoses** [comment]
> @majormoses can you review and merge now? Once this is done, I'd like to get some of the latest dependencies brought in if they aren't addressed here and then work to add Bonsai support.

Merged, do you want me to cut a release or do you got it?

**Comment by nixwiz** [comment]
Can you go ahead and cut the release, I'll pick up what I want to do with this next week.  Thanks!

**Comment by majormoses** [comment]
Ya I will get to it over the weekend.

**Pull request merged**